ftw/linkchecker/linkchecker.py

```diff
@@ -0,0 +1,100 @@
+from multiprocessing import Pool
+import time
+import requests
+import urls_testing
+import os
+import sys
+
+urls = urls_testing.urls
+number_of_processes = len(urls)
+timeout = 1
+
+
+def millis():
+    return int(round(time.time() * 1000))
+
+
+def get_uri_responses(urls):
+    headers = {
+        'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/50.0.2661.102 Safari/537.36'
+    }
+    error = None
+    response = None
+    start_time = millis()
+    try:
+        response = requests.head(urls.get('destination'),
+                                 timeout=timeout,
+                                 headers=headers,
+                                 allow_redirects=False)
+    except requests.exceptions.Timeout:
+        error = 'Timeout'
+    except requests.exceptions.TooManyRedirects:
+        error = 'Too many redirects'
+    except requests.exceptions.ConnectionError:
+        error = 'Connection Error'
+    except requests.exceptions.RequestException as e:
+        error = e
+    time = millis() - start_time
+    result = {
+        'origin': urls.get('origin'),
+        'destination': urls.get('destination'),
+        'data': response,
+        'error': error,
+    }
+    result.update({'time': time})
+
+    return result
+
+
+def limit_cpu():
+    try:
+        sys.getwindowsversion()
+    except AttributeError:
+        isWindows = False
+    else:
+        isWindows = True
+
+    if isWindows:
+        import win32api
+        import win32process
+        import win32con
+
+        pid = win32api.GetCurrentProcessId()
+        handle = win32api.OpenProcess(win32con.PROCESS_ALL_ACCESS, True, pid)
+        win32process.SetPriorityClass(
+            handle, win32process.BELOW_NORMAL_PRIORITY_CLASS)
+    else:
+        os.nice(19)
+
+
+def work_through_urls():
+    pool = Pool(number_of_processes, limit_cpu)
```

Comment:
You probably don't want a arbitrary number of processes, for example when I test 100 urls it should not spawn 100 processes.
I probably would not set the processes at all and use the default value `cpu_count()`.
So:
```python
    pool = Pool(initializer=limit_cpu)
```

Reply:
I'm not quite sure if I understood correctly, but since this process here is most likely I/O bound, wouldn't it make the process a lot faster if I worked with more processes than cpu's?

Reply:
I'm not sure about the impact. let's just move ahead and we'll see.

`cpu_count()*10` may still block the system.

Reply:
Ok for now I'll leave it like that then (with default -> cpu_count())